src/ai/llm/services/anthropic.ts

```diff
@@ -81,8 +109,15 @@ export class AnthropicService implements ILLMService {
                 logger.debug(`Iteration ${iterationCount}`);
 
                 // Get formatted messages from message manager
-                const messages = this.messageManager.getFormattedMessages();
-                const systemPrompt = this.messageManager.getFormattedSystemPrompt();
+                let messages = this.messageManager.getFormattedMessages();
+                // Ensure all message content fields are arrays of content blocks
+                messages = messages.map((msg) => {
+                    if (typeof msg.content === 'string') {
+                        return { ...msg, content: [{ type: 'text', text: msg.content }] };
+                    }
+                    return msg;
+                });
+                const systemPrompt = await this.messageManager.getFormattedSystemPrompt();
```

Comment:
do we really need this check for message content fields being arrays of content blocks?